### PR TITLE
test: restore polling for stabilizing deployment tests

### PR DESCRIPTION
**Pull request by 0xmichalis**
Discussed in https://github.com/kubernetes/kubernetes/commit/886052c2252925a5b2cf8bcfc483b711316930e3#commitcomment-20081416

@rmmh @janetkuo @wojtek-t ptal

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -3137,7 +3137,7 @@ func WaitForDeploymentRevisionAndImage(c clientset.Interface, ns, deploymentName
 	var deployment *extensions.Deployment
 	var newRS *extensions.ReplicaSet
 	var reason string
-	err := wait.PollImmediate(Poll, 1*time.Minute, func() (bool, error) {
+	err := wait.Poll(Poll, 1*time.Minute, func() (bool, error) {
```
Comment (mfojtik) [inline test/e2e/framework/util.go]: can we instead of polling watch the events long term?

Comment (kargakis) [inline test/e2e/framework/util.go]: We don't want to use events in tests 
https://github.com/kubernetes/kubernetes/issues/36453

**Comment by mfojtik** [comment]
LGTM

**Comment by kargakis** [comment]
Marking as P0 since this will reduce the times https://github.com/kubernetes/kubernetes/issues/32947 happens.

**Comment by kargakis** [comment]
https://github.com/kubernetes/kubernetes/pull/36352 wasn't cherry-picked in 1.5 so this should not be cherry-picked too.

**Comment by kargakis** [comment]
@k8s-bot gce etcd3 e2e test this https://github.com/kubernetes/kubernetes/issues/27680

File: test/e2e/framework/util.go (hunk 1 above)
Comment (krousey) [inline test/e2e/framework/util.go]: To be fair, my complaint was about using the messages in the events. If we enhanced events to have API stable event codes and data in addition to messages, then using events would be preferred.

**Pull request merged**